cilium/CFP-33716-distributed-packet-tracing.md

```diff
@@ -0,0 +1,598 @@
+# CFP-2024-06-12: Packet Tracing
+
+## Meta
+**SIG:** SIG-Hubble, SIG-Datapath  
+**Sharing:** Public  
+**Begin Design Discussion:** Jun 12, 2024  
+**End Design Discussion:**  
+**Cilium Release:**  
+**Authors:** Ben Bigdelle <bigdelle@google.com>, Mark St. John <markstjohn@google.com> based on the work of Nathan Perkins <nathanperkins@google.com>  
+**Reviewers:**  
+
+## Summary
+We aim to enhance cross-cluster and multi-system observability for flows that contain an `ip_trace_id` saved into a defined IP Option (referred to as a tagged flow), thus supporting tracing across networks. These tagged flows will trigger adding the ID value to Flow events messages and client (Hubble / Cilium Monitor) representation. This allows correlating flow events across systems which integrate with the span observability.
+
+Additionally, within Cilium, these flows will generate datapath debugging events regardless of the global debug-verbose setting. This will improve debuggability of production systems.
```

Comment:
How would you prevent someone from flooding the ring buffers of Cilium agents by sending packets with the right option?

Reply:
This feature is intended to be enabled only for specific debugging scenarios, so its use is controlled by system administrators. When this is disabled, the event emission functions as it does right now, and there will be no way to abuse this system.

When this feature is active, standard rate limiting still applies to tagged flows, which prevents flooding the event ring buffers.

Reply:
This would necessitate a separate buffer for traced events, otherwise the potentially disproportionate number of debug events produced by traced packets may starve other events (which would reduce the effectiveness of downstream consumers).

If Paul concurs, that should be called out in the doc (including the associated memory cost).

Reply:
Updated to specifically call this out.

Reply:
> This feature is intended to be enabled only for specific debugging scenarios, so its use is controlled by system administrators. When this is disabled, the event emission functions as it does right now.

OK, got it. That makes sense. I had not realized this would be transiently enabled for debugging.

(Considering that, you might want to make this an option that can be enabled at runtime without an agent restart, like the policy verdict filters or audit mode.)

Reply:
Thanks @pchaigno! I take it you mean via the Read-write configuration? i.e.`cilium config ...` command (using your example of audit mode: `cilium config PolicyAuditMode=Enabled`).

If my understanding is correct, that is a great suggestion.